lib/ansible/playbook/base.py

```diff
@@ -40,13 +40,17 @@
 
 BASE_ATTRIBUTES = {}
 
+class Undefined:
+    def __len__(self):
```

Comment:
so currently we have many `is None` comparisons, should this inherit from NoneType?
```
from types import NoneType

class Undefined(NoneType):
```